packages/flutter_tools/lib/src/android/gradle_utils.dart

```diff
@@ -63,89 +71,61 @@ class GradleUtils {
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
 distributionUrl=https\\://services.gradle.org/distributions/gradle-$gradleVersion-all.zip
-''', flush: true,
-      );
-    }
+''';
+    propertiesFile.writeAsStringSync(propertyContents);
   }
 }
-const String _defaultGradleVersion = '6.7';
-
-final RegExp _androidPluginRegExp = RegExp(r'com\.android\.tools\.build:gradle:(\d+\.\d+\.\d+)');
 
 /// Returns the Gradle version that the current Android plugin depends on when found,
 /// otherwise it returns a default version.
 ///
 /// The Android plugin version is specified in the [build.gradle] file within
 /// the project's Android directory.
-String getGradleVersionForAndroidPlugin(Directory directory) {
+String getGradleVersionForAndroidPlugin(Directory directory, Logger logger) {
   final File buildFile = directory.childFile('build.gradle');
   if (!buildFile.existsSync()) {
-    globals.printTrace('$buildFile doesn\'t exist, assuming AGP version: $_defaultGradleVersion');
+    logger.printTrace('$buildFile doesn\'t exist, assuming AGP version: $_defaultGradleVersion');
     return _defaultGradleVersion;
   }
   final String buildFileContent = buildFile.readAsStringSync();
   final Iterable<Match> pluginMatches = _androidPluginRegExp.allMatches(buildFileContent);
   if (pluginMatches.isEmpty) {
-    globals.printTrace('$buildFile doesn\'t provide an AGP version, assuming AGP version: $_defaultGradleVersion');
+    logger.printTrace('$buildFile doesn\'t provide an AGP version, assuming AGP version: $_defaultGradleVersion');
     return _defaultGradleVersion;
   }
   final String androidPluginVersion = pluginMatches.first.group(1);
-  globals.printTrace('$buildFile provides AGP version: $androidPluginVersion');
+  logger.printTrace('$buildFile provides AGP version: $androidPluginVersion');
   return getGradleVersionFor(androidPluginVersion);
 }
 
-const int _kExecPermissionMask = 0x49; // a+x
-
-/// Returns [true] if [executable] has all executable flag set.
-bool _hasAllExecutableFlagSet(File executable) {
-  final FileStat stat = executable.statSync();
-  assert(stat.type != FileSystemEntityType.notFound);
-  globals.printTrace('${executable.path} mode: ${stat.mode} ${stat.modeString()}.');
-  return stat.mode & _kExecPermissionMask == _kExecPermissionMask;
-}
-
-/// Returns [true] if [executable] has any executable flag set.
-bool _hasAnyExecutableFlagSet(File executable) {
-  final FileStat stat = executable.statSync();
-  assert(stat.type != FileSystemEntityType.notFound);
-  globals.printTrace('${executable.path} mode: ${stat.mode} ${stat.modeString()}.');
-  return stat.mode & _kExecPermissionMask != 0;
-}
-
-/// Gives execute permission to [executable] if it doesn't have it already.
-void _giveExecutePermissionIfNeeded(File executable) {
-  if (!_hasAllExecutableFlagSet(executable)) {
-    globals.printTrace('Trying to give execute permission to ${executable.path}.');
-    globals.os.makeExecutable(executable);
-  }
-}
-
```

Comment:
Context at https://github.com/flutter/flutter/pull/46748. \cc @blasten

Reply:
We also set the executable bit in create.dart, but that has a separate method for doing so. I can handle that cleanup in a separate PR